shared/archive_linux.go

```diff
@@ -76,6 +76,40 @@ func Unpack(file string, path string, blockBackend bool, runningInUserns bool, t
 
 	err = RunCommandWithFds(reader, nil, command, args...)
 	if err != nil {
+		// Check if squashfs is complaining about /dev files.
+		if command == "unsquashfs" {
+			// Check if we can process the error.
+			runError, ok := err.(RunError)
+			if !ok || runError.Stderr == "" {
+				return err
+			}
+
+			// Confirm that all errors are related to character or block devices.
+			found := false
+			for _, line := range strings.Split(runError.Stderr, "\n") {
+				line = strings.TrimSpace(line)
+				if line == "" {
+					continue
+				}
+
+				if !strings.Contains(line, "failed to create block device") {
```

Comment:
@stgraber why are these safe to skip if we cant unpack them? Thanks

Reply:
And could you add a comment here explaining why that is for future reference.

Reply:
Same reason we filter out all /dev/* stuff above in the same function. We don't care about them and can't create them in unpriv containers.

Reply:
@stgraber ah ok, I didn't see that part, but looking now, does that mean this skipping should only be done if `runningInUserns` is true like above? That would help to self-document it too.